internal/controller/stages/stages_test.go

```diff
@@ -354,13 +352,7 @@ func TestSyncNormalStage(t *testing.T) {
 			},
 			reconciler: &reconciler{
 				hasNonTerminalPromotionsFn: noNonTerminalPromotionsFn,
-				checkHealthFn: func(
-					context.Context,
-					kargoapi.FreightReference,
-					[]kargoapi.ArgoCDAppUpdate,
-				) *kargoapi.Health {
-					return nil
-				},
+				appHealth:                  &mockAppHealthEvaluator{},
```

Comment:
I know you've argued in favor of this approach for some time. I maintain that there's little _technical_ difference between injecting a function directly vs injecting an implementation of an interface that _has_ only one function...... but I'll concede that this is a helluva lot easier on the eyes. 👍

Reply:
To play advocate of the devil against myself, the way the other function injections could be made easier to the eyes is by creating a mocking function wrapper. E.g.

```go
func mockFunc(expectedReturn string) func(context.Context, ...) {
    return func(context.Context, ...) {
      return expectedReturn
    }
}
```

but at that point... you are almost creating the same LOC as doing mock implementations.